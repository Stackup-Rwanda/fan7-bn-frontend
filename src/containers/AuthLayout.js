/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import Logo from '../components/logo/logo';
import './authLayout.scss';
import SocialButtons from '../components/SocialButtons';

// eslint-disable-next-line react/prop-types
export default function AuthLayout({ title, redirectMsg, redirect, redirectLocation, children }) {
  const page = redirectLocation == ' Signup' ? 'log in' : 'sign up';
  return (
    <div className="container">
      <div className="tab1">
        <Logo />

        <h6 className="title">{title}</h6>
        {children}
        <SocialButtons page={page} />
        <div className="bottom_link">
          <p href="#" className="link">
            {redirectMsg}
            <a onClick={redirect}>
              {redirectLocation}
            </a>
          </p>
        </div>
      </div>
      <div className="tab2">
        <div className="glass">
          <div className="travel__tab">
            <h1 className="travel_info_title">Travel The World</h1>
            <p className="travel_info">We make travel and accommodation</p>
            <p className="travel_info"> easy and convenient</p>
          </div>
        </div>
      </div>
    </div>
  );
}
