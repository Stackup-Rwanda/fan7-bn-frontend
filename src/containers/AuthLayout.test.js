import React from 'react';
import { shallow } from 'enzyme';
import AuthLayout from './AuthLayout';

describe('Auth Layout component test', () => {
    it('should return false if component does not exists', () => {
        const p = ()  => 'p';
        const wrapper = shallow(<AuthLayout />);
        expect(wrapper.find(p).exists()).toBe(false);
    });
});

describe('<AuthLayout /> shallow rendering tests', () => {
    it('matches the snapshot', () => {
      const tree = shallow(<AuthLayout />);
      expect(tree).toMatchSnapshot();
    });
  });