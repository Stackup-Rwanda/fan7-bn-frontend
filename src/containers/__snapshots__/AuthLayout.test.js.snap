// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLayout /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="tab1"
  >
    <Logo />
    <h6
      className="title"
    />
    <SocialButtons
      page="sign up"
    />
    <div
      className="bottom_link"
    >
      <p
        className="link"
        href="#"
      >
        <a />
      </p>
    </div>
  </div>
  <div
    className="tab2"
  >
    <div
      className="glass"
    >
      <div
        className="travel__tab"
      >
        <h1
          className="travel_info_title"
        >
          Travel The World
        </h1>
        <p
          className="travel_info"
        >
          We make travel and accommodation
        </p>
        <p
          className="travel_info"
        >
           easy and convenient
        </p>
      </div>
    </div>
  </div>
</div>
`;
